### PR TITLE
Update Packages

### PR DESCRIPTION
**Pull request by brian-reichle**

### CODE REVIEW

File: src/WTG.BulkAnalysis.Core/Internal/AnalyzerCache.cs [hunk 1]
```diff
@@ -107,11 +107,20 @@ static ImmutableArray<T> GetCached<T>(string assemblyName, ConcurrentDictionary<
...
-					.SelectMany(p => p.AnalyzerReferences)
-					.Select(a => a.FullPath)
-					.Distinct(StringComparer.OrdinalIgnoreCase);
+				var result = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+				foreach (var project in solution.Projects)
+				{
+					foreach (var reference in project.AnalyzerReferences)
+					{
+						if (!string.IsNullOrEmpty(reference.FullPath))
```
Comment (yaakov-h) [inline src/WTG.BulkAnalysis.Core/Internal/AnalyzerCache.cs:116]: Is the change here purely for nullability analysis?

File: src/WTG.BulkAnalysis.Core/Internal/CodeFixEquivalenceGroup.cs [hunk 2]
```diff
@@ -56,8 +57,17 @@ public static async Task<ImmutableArray<CodeFixEquivalenceGroup>> CreateAsync(Co
...
 
 					var codeActions = await GetFixesAsync(solution, codeFixProvider, diagnostic, cancellationToken).ConfigureAwait(false);
+					equivalenceKeys.Clear();
 
-					foreach (var key in codeActions.Select(x => x.EquivalenceKey).Distinct())
+					foreach (var action in codeActions)
+					{
+						if (action.EquivalenceKey != null)
+						{
+							equivalenceKeys.Add(action.EquivalenceKey);
```
Comment (yaakov-h) [inline src/WTG.BulkAnalysis.Core/Internal/CodeFixEquivalenceGroup.cs:66]: Same question here

File: src/WTG.BulkAnalysis.Core/Internal/SolutionProcessor.cs [hunk 3]
```diff
@@ -165,6 +165,12 @@ async Task<ImmutableArray<Diagnostic>> GetProjectAnalyzerDiagnosticsAsync(Projec
 				.GetCompilationAsync(context.CancellationToken)
 				.ConfigureAwait(false);
 
+			if (compilation == null)
+			{
+				context.Log.WriteFormatted($"  - Skipping {project.Name} as it does not support compilation.", LogLevel.Warning);
```
Comment (yaakov-h) [inline src/WTG.BulkAnalysis.Core/Internal/SolutionProcessor.cs:170]: That's a new one to me. What projects don't support compilation? Shared projects?

File: src/WTG.BulkAnalysis.Core/Processor.cs [hunk 4]
```diff
@@ -119,7 +119,17 @@ static async Task<Workspace> LoadSolutionIntoNewWorkspace(RunContext context, st
...
 		static void ConfigureWorkspace(Workspace workspace)
 		{
-			workspace.Options = workspace.Options
+			Solution solution;
+
+			do
+			{
+				solution = workspace.CurrentSolution;
+			}
+			while (!workspace.TryApplyChanges(ConfigureSolution(solution)));
```
Comment (yaakov-h) [inline src/WTG.BulkAnalysis.Core/Processor.cs:128]: Why the change here to loop until it sticks?

File: src/WTG.BulkAnalysis.Core/Internal/AnalyzerCache.cs (hunk 1 above)
Comment (brian-reichle) [reply src/WTG.BulkAnalysis.Core/Internal/AnalyzerCache.cs:116]: Yes.

File: src/WTG.BulkAnalysis.Core/Internal/CodeFixEquivalenceGroup.cs (hunk 2 above)
Comment (brian-reichle) [reply src/WTG.BulkAnalysis.Core/Internal/CodeFixEquivalenceGroup.cs:66]: Yes, again.

File: src/WTG.BulkAnalysis.Core/Internal/SolutionProcessor.cs (hunk 3 above)
Comment (brian-reichle) [reply src/WTG.BulkAnalysis.Core/Internal/SolutionProcessor.cs:170]: Shared projects won't even load (complains about not being able to identify the language from the `.shproj` file extension. This change is mostly because `.GetCompilationAsync` is now decorated as returning a nullable compilation and the [docs](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.project.getcompilationasync?view=roslyn-dotnet-3.9.0#returns) say that it can return null when `SupportsCompilation` returns false. The [doc's for SupportsCompilation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.project.supportscompilation?view=roslyn-dotnet-3.9.0#Microsoft_CodeAnalysis_Project_SupportsCompilation) don't really provide any more detail on that.

Comment (yaakov-h) [reply src/WTG.BulkAnalysis.Core/Internal/SolutionProcessor.cs:170]: Interesting, something to keep an eye out for in future then.

File: src/WTG.BulkAnalysis.Core/Processor.cs (hunk 4 above)
Comment (brian-reichle) [reply src/WTG.BulkAnalysis.Core/Processor.cs:128]: The docs say it can return false if the new solution is not based on the latest solution. It shouldn't really happen since we set this up fairly soon after loading the solution and before we pass it any other threads ... but the alternative would be to ignore the result.

Comment (yaakov-h) [reply src/WTG.BulkAnalysis.Core/Processor.cs:128]: If it shouldn't happen then I'd be tempted to just whack a `Debug.Assert` on the boolean.

**Review by yaakov-h — APPROVED** [review]

File: src/WTG.BulkAnalysis.Core/Processor.cs (hunk 4 above)
Comment (brian-reichle) [reply src/WTG.BulkAnalysis.Core/Processor.cs:128]: maybe log it as an error?
